File: app/src/main/java/com/mahesh/mvpsample/detail/DetailContract.java
package com.mahesh.mvpsample.detail;

import com.mahesh.mvpsample.BasePresenter;
import com.mahesh.mvpsample.detail.adapter.DetailAdapterModel;

import java.util.List;

public class DetailContract {

    interface View {
        void notifyAdapterEntityChanged();

        void showToast(String text);
    }

    public interface AdapterPresenter {
        DetailAdapterModel getAdapterEntity(int position);

        int getAdapterEntityCount();

        void adapterEntityClicked(int position);
    }

    interface Presenter extends BasePresenter {

        void resume(View view);

        void pause();
    }

    interface ModelInteractor {
        void createAdapterEntityList(CreateDataCallback callback);

        List<DetailAdapterModel> getAdapterEntityList();
    }

    interface CreateDataCallback {
        void onDataCreated();
    }
}

File: app/src/main/java/com/mahesh/mvpsample/detail/DetailContract.kt
package com.mahesh.mvpsample.detail

import com.mahesh.mvpsample.BasePresenter
import com.mahesh.mvpsample.detail.adapter.DetailAdapterModel

class DetailContract {
    interface View {
        fun notifyAdapterEntityChanged()
        fun showToast(text: String?)
    }

    interface AdapterPresenter {
        fun getAdapterEntity(position: Int): DetailAdapterModel
        val adapterEntityCount: Int
        fun adapterEntityClicked(position: Int)
    }

    internal interface Presenter : BasePresenter {
        fun resume(view: View?)
        fun pause()
    }

    internal interface ModelInteractor {
        fun createAdapterEntityList(callback: () -> Unit)
        val adapterEntityList: List<DetailAdapterModel>
    }
}
